llvm/include/llvm/Transforms/Vectorize/LoopVectorizationLegality.h

```diff
@@ -269,6 +269,12 @@ class LoopVectorizationLegality {
   /// induction descriptor.
   using InductionList = MapVector<PHINode *, InductionDescriptor>;
 
+  /// ConditionalScalarAssignmentList contains the
+  /// ConditionalScalarAssignmentDescriptors for all the conditional scalar
+  /// assignments  that were found in the loop, rooted by their phis.
+  using ConditionalScalarAssignmentList =
+      MapVector<PHINode *, ConditionalScalarAssignmentDescriptor>;
```

Comment:
Why is this a MapVector? I don't think we use the lookup functionality, so a simple SmallVector of std::pair would suffice?

Reply:
`isConditionalScalarAssignmentPhi` is using the lookup functionality. Using a SmallVector here would be bad for performance since we'd have to iterate through to find Phi.

This data structure is analogous to `ReductionList` and `InductionList` which are also using MapVector.